components/core/src/package/archive.rs

```diff
@@ -417,6 +419,95 @@ pub trait FromArchive: Sized {
     fn from_archive(archive: &mut PackageArchive) -> result::Result<Self, Self::Error>;
 }
 
+// Exposes the extended archive and header metadata. Types are stored as non habitat primitives
+// with the intent being ease of deserialization into content such as conveniently formatted json
+// at the client display layer.
+#[derive(Serialize)]
+pub struct PackageArchiveInfo {
+    pub format_version: String,
+    pub key_name:       String,
+    pub hash_type:      String,
+    pub signature_raw:  String,
+    pub origin:         String,
+    pub name:           String,
+    pub version:        String,
+    pub release:        String,
+    pub checksum:       String,
+    pub target:         String,
+    pub is_a_service:   bool,
+    pub deps:           Vec<String>,
+    pub tdeps:          Vec<String>,
+    pub build_deps:     Vec<String>,
+    pub build_tdeps:    Vec<String>,
+    pub exposes:        Vec<u16>,
+    pub manifest:       String,
+    pub config:         Option<String>,
+    pub svc_user:       Option<String>,
+    pub ld_run_path:    Option<String>,
+    pub ldflags:        Option<String>,
+    pub cflags:         Option<String>,
+}
+
+impl TryFrom<PackageArchive> for PackageArchiveInfo {
+    type Error = Error;
+
+    fn try_from(mut archive: PackageArchive) -> Result<Self> {
+        let header = artifact::get_artifact_header(&archive.path)?;
+        let ident = archive.ident()?;
+        if ident.fully_qualified() {
+            Ok(PackageArchiveInfo { format_version: header.format_version,
+                                    key_name:       header.key_name,
+                                    hash_type:      header.hash_type,
+                                    signature_raw:  header.signature_raw,
+                                    origin:         ident.origin.clone(),
+                                    name:           ident.name.clone(),
+                                    version:        ident.version.unwrap_or_default(),
+                                    release:        ident.release.unwrap_or_default(),
```

Comment:
One more thing... displaying the dependency idents as their full identifier in string form (i.e., `core/foo/1.0.0/20200722110400`) is great, but it would also be nice to add an `ident` field here that does the same for the package we're providing information for.

Reply:
I have added these changes in https://github.com/habitat-sh/habitat/pull/7814/commits/5e1bee29f1f984a482a0cccef3b1819e8fc1e240